### PR TITLE
George roumieh-w1-usingAPI

### PR DESCRIPTION
**Pull request by george2621**

### CODE REVIEW

File: 3-UsingAPIs/Week1/homework/ex1-johnWho.js [hunk 1]
```diff
@@ -8,21 +8,24 @@ Rewrite this function, but replace the callback syntax with the Promise syntax:
...
-    if (!firstName) {
-      callback(new Error("You didn't pass in a first name!"));
-      return;
+
+
+const getAnonName = (firstName) => {
+  return new Promise((resolve, reject) => {
+    const fullName = firstName + ' Doe';
+    if(firstName){
+      resolve(fullName);
```
Comment (jontra000) [inline 3-UsingAPIs/Week1/homework/ex1-johnWho.js:17]: Style note: Best practise is to move the declaration of fullName here. Keeping the declaration of variables as close as possible to where they are used makes the code cleaner and easier to understand, and reduces the chances of introducing bugs.

File: 3-UsingAPIs/Week1/homework/ex3-rollDice.js [hunk 2]
```diff
@@ -11,47 +11,34 @@ Full description at: https://github.com/HackYourFuture/Homework/tree/main/3-Usin
...
-// TODO Refactor to use promise
-rollDice((error, value) => {
-  if (error !== null) {
-    console.log(error.message);
-  } else {
-    console.log(`Success! Dice settled on ${value}.`);
-  }
-});
-
-// ! Do not change or remove the code below
```
Comment (jontra000) [inline 3-UsingAPIs/Week1/homework/ex3-rollDice.js:56]: The unit tests are failing because these lines have been removed

**Review by jontra000 — CHANGES_REQUESTED** [review]
Hi @george2621, the code all looks good and works. Nice work there. However you still need to add your explanations for the behaviour in ex3 and ex4 as requested in the homework questions before I can close as approved

**Comment by george2621** [comment]
@jontra000 Thanks, I will fix it asap

**Comment by george2621** [comment]
@jontra000 I fixed all issues but when I  try to make re-request isn't working

**Review by jontra000 — APPROVED** [review]
Thanks @george2621. All looks good now

**Comment by george2621** [comment]
Thanks  for reviewing and advising @jontra000

**Comment by george2621** [comment]
@can you approve it now ? or does it needs more modification?

**Comment by jontra000** [comment]
No more modifications needed. I'm still waiting for write permissions on the repo to approve it but I will as soon as I get them.

**Pull request closed without merging**